test/logflare/backends/adaptor/axiom_adaptor_test.exs

```diff
@@ -0,0 +1,302 @@
+             """
+             |> Jason.decode!()
+         }}
+      end)
+
+      assert :ok = @subject.test_connection(ctx.backend)
+      assert :ok = @subject.test_connection({ctx.source, ctx.backend})
+    end
+
+    test "returns error on 403 response", ctx do
+      @tesla_adapter
+      |> expect(:call, fn _env, _opts ->
+        {:ok,
+         %Tesla.Env{
+           status: 403,
+           body:
+             """
+             {
+               "code": 403,
+               "message": "Forbidden"
+             }
+             """
+             |> Jason.decode!()
+         }}
+      end)
+
+      assert {:error, reason} = @subject.test_connection(ctx.backend)
+      assert reason =~ "auth"
+    end
+
+    test "returns error on 404 response", ctx do
+      @tesla_adapter
+      |> expect(:call, fn _env, _opts ->
+        {:ok,
+         %Tesla.Env{
+           status: 404,
+           body:
+             """
+             {
+               "code": 404,
+               "message": "Not found"
+             }
+             """
+             |> Jason.decode!()
+         }}
+      end)
+
+      assert {:error, reason} = @subject.test_connection(ctx.backend)
+      assert reason =~ ctx.config.dataset_name
+    end
+
+    test "returns error on 500 response", ctx do
+      @tesla_adapter
+      |> expect(:call, fn _env, _opts ->
+        {:ok, %Tesla.Env{status: 500, body: ""}}
+      end)
+
+      assert {:error, reason} = @subject.test_connection(ctx.backend)
+      assert is_binary(reason)
+    end
+
+    test "returns error on request error", ctx do
+      @tesla_adapter
+      |> expect(:call, fn _env, _opts ->
+        {:error, :nxdomain}
+      end)
+
+      assert {:error, :nxdomain} = @subject.test_connection(ctx.backend)
+    end
+  end
+
+  describe "logs ingestion" do
+    setup :backend_data
+
+    setup %{source: source, backend: backend} do
+      start_supervised!({AdaptorSupervisor, {source, backend}})
+      :timer.sleep(250)
+      :ok
+    end
+
+    test "sends logs via REST API", %{source: source, config: config} do
+      this = self()
+      ref = make_ref()
+
+      Client
+      |> expect(:send, fn opts ->
+        body = opts[:body]
+
+        assert opts[:headers]["content-type"] == "application/json"
+        assert opts[:headers]["authorization"] == "Bearer #{config.api_token}"
+
+        send(this, {ref, body})
+        %Tesla.Env{status: 200, body: ""}
+      end)
+
+      message = "Test log message"
+
+      log_events = [
+        build(:log_event,
+          source: source,
+          event_message: message,
+          random_attribute: "nothing",
+          timestamp: 1_704_067_200_000_000
+        )
+      ]
+
+      assert {:ok, _} = Backends.ingest_logs(log_events, source)
+      assert_receive {^ref, body}, 2000
+      assert [log] = body
+      assert log["event_message"] == message
+      assert log["timestamp"] == "2024-01-01T00:00:00.000000Z"
+      assert log["random_attribute"] == "nothing"
```

Comment:
```suggestion
      assert_receive {^ref, [log]}, 2000
      assert log["event_message"] == log_event.body["event_message"]
      assert log["timestamp"] != log_event.body["timestamp"]
      assert log["random_attribute"] == "nothing"
```

Reply:
1. `log_event.body["event_message"]` strongly couples this test to the log_event shape. Can I assume this won't change any time soon?
2. This is the only test verifying the correctness of timestamp conversion. Why do you suggest to remove it?

Reply:
- event_message won't change and is a special field and will always be present. 
- removes need for specific timestamp being set above and reducing loc, and asserts the same thing (that the unix timestamp gets formatted). Testing that it's an ISO timestamp doesn't matter as it's an implementation detail.

Reply:
> Testing that it's an ISO timestamp doesn't matter as it's an implementation detail.

On that I wouldn't agree, the timestamp is parsed by Axiom, so it has to conform to the format passed as a query parameter, the fact it has ISO format with particular precision is part of the contract I'd like to test here